tests/test_readwrite.py

```diff
@@ -985,3 +980,19 @@ def test_write_elem_consolidated(tmp_path: Path):
         ValueError, match="Cannot overwrite/edit a store with consolidated metadata"
     ):
         ad.io.write_elem(g["obs"], "foo", np.arange(10))
+
+
+@pytest.mark.zarr_io
+def test_write_elem_version_mismatch(tmp_path: Path):
+    if is_zarr_v2():
```

Comment:
please use `@pytest.mark.skipif` instead of imperative skipping whenever possible (e.g. very much here)